servicetalk-buffer-api/src/main/java/io/servicetalk/buffer/api/CharSequences.java

```diff
@@ -206,39 +206,81 @@ public static int caseInsensitiveHashCode(CharSequence seq) {
     }
 
     /**
-     * Split a given {@link CharSequence} to separate ones on the given {@code delimiter}.
-     * The returned {@link CharSequence}s are created by invoking the {@link CharSequence#subSequence(int, int)} method
-     * on the main one.
+     * Split a given {@link #newAsciiString(Buffer) AsciiString} to separate ones on the given {@code delimiter}.
      *
+     * Trimming white-space before and after each token can be controlled by the {@code trim} flag
      * This method has no support for regex.
      *
-     * @param input The initial {@link CharSequence} to split, this experiences no side effects
-     * @param delimiter The delimiter character
+     * @param input The initial {@link CharSequence} to split, this experiences no side effects.
+     * @param delimiter The delimiter character.
+     * @param trim Flag to control whether the individual items must be trimmed.
      * @return a {@link List} of {@link CharSequence} subsequences of the input with the separated values
      */
-    public static List<CharSequence> split(final CharSequence input, final char delimiter) {
+    public static List<CharSequence> split(final CharSequence input, final char delimiter, final boolean trim) {
         if (input.length() == 0) {
             return emptyList();
         }
 
+        boolean isAscii = isAsciiString(input);
+        List<CharSequence> result = new ArrayList<>(4);
+
         int startIndex = 0;
-        List<CharSequence> result = new ArrayList<>();
-        for (int i = 0; i < input.length(); i++) {
-            if (input.charAt(i) == delimiter) {
-                if ((i - startIndex) > 0) {
-                    result.add(input.subSequence(startIndex, i));
+        int endIndex = -1;
+
+        if (trim) {
+            startIndex = -1;
+            for (int i = 0; i < input.length(); i++) {
+                char c = input.charAt(i);
+                if (c != ' ' && c != delimiter) {
+                    endIndex = i + 1;
                 }
 
-                startIndex = i + 1;
+                if (startIndex == -1 && c != ' ' && c != delimiter) {
+                    startIndex = i;
+                } else if (c == delimiter) {
+                    if (endIndex > startIndex) {
+                        result.add(subsequence(isAscii, input, startIndex, endIndex));
+                    } else {
+                        result.add(isAscii ? newAsciiString("") : "");
```

Comment:
Can you comment if the following unit test is correct? If so this code needs to be adjusted.
```java
// there are no non-delimiter chars present -> content should be empty
assertThat(split("  ", ' ', true), empty());
```

we may need more state (e.g. differentiate between last index of non-space and delimiter)?

Reply:
Not the case anymore behavior is as described in comment above.

- `split("  ", ' ', true) -> ["", "", ""]`

similar to
- `split(",,", ' ', true) -> ["", "", ""]`